core/kernel/interrupt.c

```diff
@@ -42,7 +44,17 @@ int dt_get_irq(const void *fdt, int node)
 	if (!prop)
 		return it_num;
 
-	return itr_chip->dt_get_irq(prop, len);
+	if (!type)
+		type = &ltype;
+	else
+		*type = IRQ_TYPE_NONE;
```

Comment:
Since I added the possibility of passing NULL pointers in dt_get_irq callback, I removed the intermediate variables and kept only the initialization part (Or do you want it to be removed also ?).

Reply:
No need to set a default value here IMO.